grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1083,12 +1087,109 @@ class StockMovementService {
+            }
+            if (params.lot?.contains("E") && NumberUtils.isNumber(params.lot)) {
+                params.errors.push("Lot numbers must not be specified in scientific notation. " +
+                        "Please reformat field with Lot Number: \"${params.lot}\" to a number format")
+            }
+
+            PickPageItem pickPageItem = pickPageItems.find { it.requisitionItem?.id == params.id }
+            if (!pickPageItem) {
+                params.errors.push("Requisition item id: ${params.id} not found")
+            }
+            if (pickPageItem) {
+                AvailableItem availableItem = pickPageItem.availableItems?.find { item ->
+                    Boolean binLocationMatches = params.binLocation ? item.binLocation?.name == params.binLocation : !item.binLocation
+                    Boolean lotMatches = params.lot == item.inventoryItem?.lotNumber
+                    binLocationMatches && lotMatches
+                }
+
+                if (!availableItem) {
+                    params.errors.push("There is no item available with: ${params.lot ? "lot ${params.lot}" : ""} ${params.binLocation ? "bin ${params.binLocation}" : ""}")
+                }
+            }
+
+            Integer itemQuantitySum = groupedItems[params.id].sum { Integer.parseInt(it.quantity) }
+            if (pickPageItem && itemQuantitySum > pickPageItem.requisitionItem.quantity) {
+                if (!supportsOverPick) {
+                    params.errors.push("Item ${params.id} is overpicked, expected quantity ${pickPageItem.requisitionItem.quantity}, received ${itemQuantitySum}")
+                }
+            } else if (itemQuantitySum != pickPageItem.requisitionItem.quantity) {
+                params.errors.push("Item ${params.id} expected quantity is ${pickPageItem.requisitionItem.quantity}, received ${itemQuantitySum}")
+            }
+        }
+    }
+
+    void importPicklistTemplate(List<Map> data, StockMovement stockMovement, List<PickPageItem> pickPageItems) {
+        data.each { params ->
+            // skip rows with errors
+            if (params.errors) {
+                return
+            }
+
+            PickPageItem pickPageItem = pickPageItems.find {
+                it.requisitionItem?.id == params.id
+            }
+
+            AvailableItem availableItem = pickPageItem.availableItems?.find { item ->
+                Boolean binLocationMatches = params.binLocation ? item.binLocation?.name == params.binLocation : !item.binLocation
+                Boolean lotMatches = params.lot == item.inventoryItem?.lotNumber
+                binLocationMatches && lotMatches
+            }
+
+            RequisitionItem requisitionItem = pickPageItem.requisitionItem?.modificationItem ?: pickPageItem.requisitionItem
+
+            pickPageItem.picklistItems.each {
+                if (it.id) {
+                    it.quantity = 0
+                }
+            }
+            pickPageItem.picklistItems.add(new PicklistItem(
```

Comment:
Can we just bottle up the picklist items in their own list and pass them to the createOrUpdatePicklistItems() method? Even if that means creating a new method?